src/Map/SavedGameDataSection2.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include <cstdint>
+#include <vector>
+
+// Second section of saved game specifc data (not included in .map files)
+struct SavedGameDataSection2
+{
+	uint32_t unitCount;
+	uint32_t unknown1;
+	uint32_t unknown2;
+	uint32_t unknown3;
+	uint32_t sizeOfUnit;
+	uint32_t objectCount1;
+	uint32_t objectCount2;
+	std::vector<uint8_t> objects1;
+	std::vector<uint32_t> objects2;
+	uint32_t unitID1;
+	uint32_t unitID2;
+	std::vector<uint8_t> unitRecord;
+};
```

Comment:
I'm wondering if we should avoid storing `vector` fields in the same struct as fixed length fields. The main use of fixed length fields is serialization, which we can't do if vectors are embedded in the same struct.

----

We could create dummy structs for object1 and object2, which contain just an array of `uint8_t`. That would give a better type and automatic sizing for `objects1` and `objects2`. Same for the `unitRecord` field.

Reply:
I think line 20, unit record could just be a std::array if it is always the same fixed size which should play nicer with serialization? I'm not quite following what you are proposing for the other 2 vectors though. I'm happy leaving this alone until (if ever) we want to polish reading a saved game. Unless someone wants to research the new size of the vector unitRecord and then I could quickly update that.

Thanks,
Brett

Reply:
You're probably right about using a `std::array` for the last field, at least if we only ever want to support the newer version of the saved game files. However, with the other two fields remaining as vectors, there isn't much point. For serialization, it's an all or nothing thing.

I'd say ignore it for now. We use some fixed sized fields, but this isn't really a serialization struct. Each field is serialized individually.